src/integer_mod_q/z_q/fmpz_mod_helpers.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains helpful functions on [`fmpz`] values in a ring/`modulus` context.
+
+use crate::integer::fmpz_helpers::distance;
+use flint_sys::fmpz::{fmpz, fmpz_cmp};
+
+const ZERO_FMPZ: fmpz = fmpz(0);
+
+/// Computes the shortest distance of `self` to the next zero instance
+/// regarding the `modulus`.
+///
+/// WARNING: This function assumes `value` to be reduced,
+/// i.e. `0 <= value <= modulus`.
+///
+/// # Example
+/// ```compile_fail
+/// use flint_sys::fmpz::fmpz;
+/// use qfall_math::integer_mod_q::fmpz_mod_helpers::length;
+///
+/// let modulus = fmpz(15);
+/// let value = fmpz(10);
+///
+/// let length = length(&value, &modulus);
+///
+/// assert_eq!(5, length.0);
+/// ```
+pub(crate) fn length(value: &fmpz, modulus: &fmpz) -> fmpz {
+    let distance_zero = distance(&ZERO_FMPZ, value);
+    let distance_modulus = distance(value, modulus);
+
+    // if distance_zero < distance modulus => return distance_zero
+    if unsafe { fmpz_cmp(&distance_zero, &distance_modulus) } < 0 {
+        distance_zero
+    } else {
+        distance_modulus
```

Comment:
Memory leak, the not returned value is never cleared.

Reply:
Not a problem anymore